platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,1026 @@
+
+- Delivery of TSS-Ledger-ID will require that the TSS-Roster proposal is delivered in a prior release.
+- Development of the TSS-Ledger-ID proposal is dependent on TSS-Roster proposal in the following ways:
+  - The `Roster` data structures must be defined and stored in state.
+  - To modify the roster adoption lifecycle with TSS conditions, the TSS-Roster life-cycle must be implemented.
+
+Impacts of TSS-Ledger-ID to Services Team
+
+- The timing of submitting the candidate roster may need to be adjusted to allow enough time for the candidate
+  roster to have key material generated.
+- Services will need to know or detect when the candidate roster fails to be adopted due to not having enough key
+  material generated.
+- More than half and preferably all the nodes' public `tssEncryptionKeys` will need to be added to the address book.
+  - This requires modification of the HAPI transactions that manage the address book.
+
+Impacts to DevOps Team
+
+- Each consensus node will need a new `tssEncryptionKey` in addition to the existing RSA key.
+  - EC Key generation will have to happen before a node can join the network.
+- A new node added to an existing network will need to be given a state from an existing node after the network has
+  adopted the roster containing the new node. The new node must be in the active roster to join the network.
+- If a node is added, removed, or updated in the address book and the candidate roster
+  is not adopted in the next software upgrade, the change to that node is not applied.
+- config only software upgrades can be used to trigger the adoption of a candidate roster, if the candidate roster
+  has received enough yes votes.
+
+Implications Of Completion
+
+- The network will have a public ledger id that can verify ledger signatures.
+- The active consensus nodes can transfer the ability to sign messages to the next set of consensus nodes.
+- `TSS-Block-Signing` and `TSS-Ledger-ID-Updates` proposals becomes unblocked for development.
+
+### Requirements
+
+TSS Core Requirements
+
+| Requirement ID |                                                        Requirement Description                                                        |
+|----------------|---------------------------------------------------------------------------------------------------------------------------------------|
+| TSS-001        | The ledger MUST have a public/private signing key pair that does not change when the consensus roster changes.                        |
+| TSS-002        | The ledger private key MUST be a secret that nobody knows.                                                                            |
+| TSS-003        | The `ledger id` MUST be a public key that everyone knows and that is able to verify signatures from the ledger private key.           |
+| TSS-004        | The ledger signature MUST be produced through aggregating signatures from consensus nodes representing at least 1/3 consensus weight. |
```

Comment:
I understand the logic (assuming 2/3+1 nodes by stake weight are honest, if at least 1/3 of signatures are collected by stake weight, then we must have included at least 1 honest node). However, there are ramifications to this instead of 2/3.

I was talking with Oleg today about a way to determine dynamic network throttling. We may have a very nice way to do this, if we used 2/3 as the requirement for signing. It would mean that any given node could look at the number of rounds produced by its hashgraph that are newer than the most recent signed block/round and determine how many rounds the entire network is behind. This could be used as a deterministic measure by which the network as a whole could throttle back.

If signing works for 1/3 instead of 2/3, then this dynamic throttle idea may not work. I'm not sure at this moment what to make of it, other than, maybe we should go with 2/3 for now and adjust to 1/3 in the future if we find we can?

Reply:
For now, ignore this suggestion. Let's keep it as defined in this document and consider the throttling problem separately.

Reply:
Later on in the design doc there is a theorem that requires 1/2 shares to sign for a security guarantee in the down sampling from stake to weight to shares.